common/app/layout/Column.scala

```diff
@@ -63,20 +61,35 @@ case class HtmlAndClasses(index: Int, html: Html, classes: Seq[String])
 
 object SliceWithCards {
   def fromBlobs(layout: SliceLayout, blobs: Seq[HtmlAndClasses]): SliceWithCards = {
-    val columns = layout.columns.toList.mapAccumL(blobs) { case (itemsRemaining, column) =>
-      val (itemsForColumn, itemsNotConsumed) = itemsRemaining splitAt column.numItems
-
-      (itemsNotConsumed, ColumnAndCards(column, itemsForColumn.zipWithIndex map {
-        case (HtmlAndClasses(index, html, classes), positionInColumn) =>
-          FaciaCardAndIndex(
-            index,
-            HtmlBlob(html, classes, Column.cardStyle(column, positionInColumn).getOrElse(ItemClasses.showMore)),
-            None
-          )
-      }))
-    }._2
-
-    SliceWithCards(layout.cssClassName, columns)
+
+    @tailrec
+    def columnsWithCards(columns: List[Column],
+                         items: Seq[HtmlAndClasses],
+                         accumulation: List[ColumnAndCards] = List.empty
```

Comment:
Probably want to switch this to a `Vector` because here you have to continually append to the end of the `List` which will be slower than it will be with a `Vector`.